src/mon/FSCommands.cc

```diff
@@ -108,6 +108,18 @@ class FailHandler : public FileSystemCommandHandler
       return -ENOENT;
     }
 
+  bool confirm = false;
+  cmd_getval(cmdmap, "yes_i_really_mean_it", confirm);
+  if (!confirm &&
+      mon->mdsmon()->has_health_warnings({
+	MDS_HEALTH_TRIM, MDS_HEALTH_CACHE_OVERSIZED})) {
+    ss << "MDS has one of the two health warnings, either MDS_TRIM or "
+       << "MDS_CACHE_OVERSIZED. MDS failover isn't recommended in such cases "
+       << "since it might create problems due to slow recovery. If you wish "
+       << "to proceed, pass --yes-i-really-mean-it";
```

Comment:
```suggestion
    ss << "MDS has one of two health warnings which could extend recovery: MDS_TRIM or "
       << "MDS_CACHE_OVERSIZED. MDS failover is not recommended "
       << "since it might cause unexpected file system unavailability. If you wish "
       << "to proceed, pass --yes-i-really-mean-it";
```

Reply:
Done.